defaults.go

```diff
@@ -80,6 +90,10 @@ var defaults = []struct {
 		fallback: func(cfg *Config) bool { return cfg.Peerstore == nil },
 		opt:      DefaultPeerstore,
 	},
+	{
+		fallback: func(cfg *Config) bool { return cfg.ListenAddrs == nil },
```

Comment:
We should probably only apply these if we are using the default transports. I'd put this *before* the default transport section and check to see if `Transports` is nil.